# Imports

In [2]:
import pandas as pd

# Importing Data

In [4]:
df = pd.read_csv('House_price.csv')

In [10]:
df.head(5)

,Avg. Area Income,House Age,Number of Rooms,Number of Bedrooms,Area Population,Price,Address
0,79545.45857,5.682861,7.009188,4.09,23086.80050,1.059034e+06,"208 Michael Ferry Apt. 674\nLaurabury, NE 3701..."
1,79248.64245,6.002900,6.730821,3.09,40173.07217,1.505891e+06,"188 Johnson Views Suite 079\nLake Kathleen, CA..."
2,61287.06718,5.865890,8.512727,5.13,36882.15940,1.058988e+06,"9127 Elizabeth Stravenue\nDanieltown, WI 06482..."
3,63345.24005,7.188236,5.586729,3.26,34310.24283,1.260617e+06,USS Barnett\nFPO AP 44820
4,59982.19723,5.040555,7.839388,4.23,26354.10947,6.309435e+05,USNS Raymond\nFPO AE 09386


In [16]:
df.sample(n=5)

,Avg. Area Income,House Age,Number of Rooms,Number of Bedrooms,Area Population,Price,Address
3245,67196.12542,6.020039,8.624191,4.11,38417.98087,1.355546e+06,"38974 Tara Points Suite 917\nSouth Davidfort, ..."
2973,62315.58370,6.682463,5.764141,3.41,47163.75462,1.142264e+06,"83004 Kari Squares\nLake Rose, TX 01776"
2975,66197.67377,5.790273,5.895335,4.19,55849.46922,1.244917e+06,"16466 Mason Roads Apt. 483\nNew Danaberg, RI 1..."
3738,72083.34088,4.174159,5.728892,2.44,36471.45807,8.853798e+05,"439 Bishop Path\nLake Richardtown, NH 13156-6062"
2800,89475.55970,6.888439,5.233514,2.40,31070.43183,1.707270e+06,54863 Warner Greens Suite 781\nPort Larryville...


In [6]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 4548 entries, 0 to 4547
Data columns (total 7 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Avg. Area Income    4548 non-null   float64
 1   House Age           4548 non-null   float64
 2   Number of Rooms     4548 non-null   float64
 3   Number of Bedrooms  4548 non-null   float64
 4   Area Population     4548 non-null   float64
 5   Price               4548 non-null   float64
 6   Address             4548 non-null   str    
dtypes: float64(6), str(1)
memory usage: 248.8 KB


In [23]:
df.isnull().sum()

Avg. Area Income      0
House Age             0
Number of Rooms       0
Number of Bedrooms    0
Area Population       0
Price                 0
Address               0
dtype: int64

In [19]:
df.describe()

,Avg. Area Income,House Age,Number of Rooms,Number of Bedrooms,Area Population,Price
count,4548.000000,4548.000000,4548.000000,4548.000000,4548.000000,4.548000e+03
mean,68611.700818,5.978918,6.987646,3.981693,36187.469334,1.233916e+06
std,10686.487761,0.990850,1.006587,1.230939,9910.189915,3.545676e+05
min,17796.631190,2.644304,3.236194,2.000000,172.610686,1.593866e+04
25%,61485.150192,5.332187,6.299692,3.140000,29423.163510,9.977751e+05
50%,68817.036575,5.960872,7.002245,4.050000,36215.560985,1.234571e+06
75%,75820.741747,6.658368,7.665871,4.490000,42880.554642,1.470616e+06
max,107701.748400,9.519088,10.759588,6.500000,69592.040240,2.469066e+06


## Findings
* We got total of 4548 rows and 7 columns.
* No missing values in the dataset.
* The address column has \n (next line operator) in the address data.
* The price data needs to be converted into fixed e.g. 2nd or 3rd deciaml places.
* The house ages is well distributed between 2.6-9.5 years, where the houses are not so old, and not so new.
* Based on the Area Income, we can say that the properties lie in high income areas(75% people earn more than 60k).
* Based on the room no.s VS bedroom no.s, we can say that the properties are of Villa or Bungalow types.
* The population density is very low, which means the properties are located in semi-rural areas.
* We found that, column index 2 and 3 are needed to be converted into int data type, since no. of rooms cannot be in fractions.